modules/ossm-creating-certificates-for-multi-cluster-mesh.adoc

```diff
@@ -0,0 +1,197 @@
+====
+
+.Prerequisites
+
+* You have OpenSSL installed locally.
+
+.Procedure
+
+. Create a root certificate file named `root-ca.conf`:
++
+.Sample root certificate file
+[source,terminal]
+----
+encrypt_key = no
+prompt = no
+utf8 = yes
+default_md = sha256
+default_bits = 4096
+req_extensions = req_ext
+x509_extensions = req_ext
+distinguished_name = req_dn
+[ req_ext ]
+subjectKeyIdentifier = hash
+basicConstraints = critical, CA:true
+keyUsage = critical, digitalSignature, nonRepudiation, keyEncipherment, keyCertSign
+[ req_dn ]
+O = Istio
+CN = Root CA
+----
+
+. Create a key for the root certificate by running the following command:
++
+[source,terminal]
+----
+$ openssl genrsa -out root-key.pem 4096
+----
+
+. Create the certificate signing request by running the following command:
++
+[source,terminal]
+----
+openssl req -sha256 -new -key root-key.pem \
+  -config root-ca.conf \
+  -out root-cert.csr
+----
+
+. Create a shared root certificate by running the following command:
++
+[source,terminal]
+----
+openssl x509 -req -sha256 -days 3650 \
+  -signkey root-key.pem \
+  -extensions req_ext -extfile root-ca.conf \
+  -in root-cert.csr \
+  -out root-cert.pem
+----
+
+. Create an intermediate certificate for the East cluster by running the following command:
++
+[source,terminal]
+----
+$ openssl genrsa -out east/ca-key.pem 4096
+----
+
+. Create an OpenSSL configuration file named `intermediate.conf` for the intermediate certificate of the East cluster by copying the following sample file and saving it locally:
++
+.Sample configuration file
+[source,subs="attributes,verbatim"]
+----
+east/intermediate.conf <1>
+[ req ]
+encrypt_key = no
+prompt = no
+utf8 = yes
+default_md = sha256
+default_bits = 4096
+req_extensions = req_ext
+x509_extensions = req_ext
+distinguished_name = req_dn
+[ req_ext ]
+subjectKeyIdentifier = hash
+basicConstraints = critical, CA:true, pathlen:0
+keyUsage = critical, digitalSignature, nonRepudiation, keyEncipherment, keyCertSign
+subjectAltName=@san
+[ san ]
+DNS.1 = istiod.istio-system.svc
+[ req_dn ]
+O = Istio
+CN = Intermediate CA
+L = east
+----
+<1> In this example, the configuration file resides in the `east` directory. If your file resides in a different directory, update the syntax accordingly.
+
+. Create a key for the root certificate by running the following command:
++
+[source,terminal]
+----
+  openssl req -new -config east/intermediate.conf \
+    -key east/ca-key.pem \
+    -out east/cluster-ca.csr
+----
+
+. Create the certificate signing request by running the following command:
++
+[source,terminal]
+----  
+  openssl x509 -req -sha256 -days 3650 \
+    -CA root-cert.pem \
+    -CAkey root-key.pem -CAcreateserial \
+    -extensions req_ext -extfile east/intermediate.conf \
+    -in east/cluster-ca.csr \
+    -out east/ca-cert.pem
+----
+
+. Create a shared root certificate for both clusters by running the following command:
```

Comment:
This command creates a certificate chain (containing the intermediate and the root CA certificates) for just one cluster, not both.
```suggestion
. Create a certificate chain from the intermediate and root CA certificate for the east cluster:
```

Reply:
done